### ISSUE TITLE
Syntax highlighting broken on macOS Monterey/Apple Silicon

### ISSUE THREAD

**Issue by colepeters** [nvim-treesitter/nvim-treesitter#3680]
### Describe the bug

Syntax highlighting appears to be broken on macOS Monterey and/or Apple Silicon. I say this because, aside from the version of macOS and the processor, everything else about my configuration is the same on both computers I'm using — both machines are running Nvim 0.8.0, on the latest iTerm, with the same dotfiles.

My configuration works as expected on my older MacBook running on Intel, but my newer MacBook, with the same config, seems to produce much worse results.

For reference, here's a file on my Intel MacBook on macOS Big Sur:
<img width="1280" alt="Screen Shot 2022-10-17 at 18 35 21" src="https://user-images.githubusercontent.com/1713932/196303222-692d0de1-3fc7-4231-aa6c-38041ea13750.png">

And the same file on my M1 Pro MacBook on macOS Monterey:
<img width="1512" alt="image" src="https://user-images.githubusercontent.com/1713932/196304665-7db0ace0-64f1-4289-b0d9-cde8eb165d0b.png">

I get only a very slight difference when running `:TSDisable highlight` on the Apple Silicon MacBook:
<img width="1512" alt="image" src="https://user-images.githubusercontent.com/1713932/196304690-6803a641-099b-4077-8cb3-a87a46b2dee2.png">

### To Reproduce

1. Check syntax highlighting on macOS w/ Apple Silicon
2. Compare with syntax highlighting on macOS w/ Intel

### Expected behavior

Syntax highlighting should be the same regardless of platform.

### Output of `:checkhealth nvim-treesitter`

```markdown
nvim-treesitter: require("nvim-treesitter.health").check()
========================================================================
## Installation
  - OK: `tree-sitter` found 0.20.7 (parser generator, only needed for :TSInstallFromGrammar)
  - OK: `node` found v16.18.0 (only needed for :TSInstallFromGrammar)
  - OK: `git` executable found.
  - OK: `cc` executable found. Selected from { vim.NIL, "cc", "gcc", "clang", "cl", "zig" }
    Version: Apple clang version 14.0.0 (clang-1400.0.29.102)
  - OK: Neovim was compiled with tree-sitter runtime ABI version 14 (required >=13). Parsers must be compatible with runtime ABI.

## Parser/Features H L F I J
  - json           ✓ ✓ ✓ ✓ .
  - vim            ✓ ✓ ✓ . ✓
  - lua            ✓ ✓ ✓ ✓ ✓
  - jsdoc          ✓ . . . .
  - c              ✓ ✓ ✓ ✓ ✓
  - json5          ✓ . . . ✓
  - html           ✓ ✓ ✓ ✓ ✓
  - javascript     ✓ ✓ ✓ ✓ ✓
  - typescript     ✓ ✓ ✓ ✓ ✓
  - regex          ✓ . . . .
  - bash           ✓ ✓ ✓ . ✓
  - css            ✓ . ✓ ✓ ✓
  - tsx            ✓ ✓ ✓ ✓ ✓
  - help           ✓ . . . .
  - comment        ✓ . . . .
  - yaml           ✓ ✓ ✓ ✓ ✓

  Legend: H[ighlight], L[ocals], F[olds], I[ndents], In[j]ections
         +) multiple parsers found, only one will be used
         x) errors found in the query, try to run :TSUpdate {lang}
```

### Output of `nvim --version`

```text
NVIM v0.8.0
Build type: Release
LuaJIT 2.1.0-beta3
Compiled by brew@HMBRW-A-001-M1-004.local

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/opt/homebrew/Cellar/neovim/0.8.0/share/nvim"

Run :checkhealth for more info
```

### Additional context

Here's my nvim-treesitter config: https://github.com/colepeters/dotfiles/blob/main/config/nvim/lua/colepeters/plugins/treesitter.lua

**Comment by itepechi** [comment]
#3656 ?

**Comment by twandylue** [comment]
I have a similar problem. In a *.md file, highlighting of visual line can not display normally.
I also use macOS(Apple Silicon(M1)).

**Comment by srillia** [comment]
https://github.com/nvim-treesitter/nvim-treesitter/commit/42ab95d5e11f247c6f0c8f5181b02e816caa4a4f#comments

**Comment by clason** [comment]
Duplicate. You need to make sure the capture names are highlighted by your color scheme.

**Comment by erikw** [comment]
> You need to make sure the capture names are highlighted by your color scheme.

@clason Do I understand this correctly that this mean that all colorschemes that we want to sue with `nvim-treesitter` needs to be updated?

Do you know any colorschemes which are working with `nvim-treesitter` highlights currently? :)

**Comment by colepeters** [comment]
> Duplicate. You need to make sure the capture names are highlighted by your color scheme.

@clason Sorry, could you please reopen this? I'm using the same colour scheme on both machines, and it's optimized for Treesitter: https://github.com/EdenEast/nightfox.nvim

**Comment by itepechi** [comment]
> @ clason Sorry, could you please reopen this? I'm using the same colour scheme on both machines, and it's optimized for Treesitter: https://github.com/EdenEast/nightfox.nvim

It does not support 0.8.0 highlight groups.

Please read the comments before requesting a reopen.
Especially [this one](https://github.com/nvim-treesitter/nvim-treesitter/issues/3680#issuecomment-1281737090).

**Comment by colepeters** [comment]
@itepechi I did indeed read the comments before replying — but respectfully, as an end user, a comment linking to a discussion on a commit that I don't have any context for is pretty hard to follow or extract meaning from.

Additionally, my configuration was working correctly on 0.8.0 on my Intel MacBook, so it seemed to me that my issue was still valid. It took me spending some more time investigating the issue to realize that Treesitter itself had not been updated on my Intel MacBook; once I did that, I was able to reproduce the issue there. None of this was clear to me from the preceding comments.

I fully respect all the work you and everyone else does on Treesitter, and I realize it takes a lot of time to deal with issues, especially if the problem ends up being on the user's side, as is the case here. But a little empathy for users who might be missing context or information that you yourself may have is always appreciated, rather than presuming I'm just not reading what's been shared. I definitely don't want to waste anyone's time! :)

Thank you!

**Comment by itepechi** [comment]
Sorry if I sounded rude.
Both neovim and nvim-treesitter are under heavy development and the API will be actively changed.
If you experience any problems in the future after the update, you may be able to fix them by checking [this](https://github.com/neovim/neovim/issues/14090) issue.

**Comment by clason** [comment]
> @clason Do I understand this correctly that this mean that all colorschemes that we want to sue with `nvim-treesitter` needs to be updated?

Yes, that is correct. Actively monitoring and adapting to changes in nvim-treesitter (which is clearly marked as "experimental") is part of "tree-sitter support".

> Do you know any colorschemes which are working with `nvim-treesitter` highlights currently? :)

No, I don't. I recommend opening an issue (or better yet, a PR) at the colorscheme you already use.

> @clason Sorry, could you please reopen this? I'm using the same colour scheme on both machines, and it's optimized for Treesitter: https://github.com/EdenEast/nightfox.nvim

No, I will not reopen this, because this is not an issue in this project that needs to be fixed; it's an intentional breaking change (valid complaints about how this change could have been handled with more resources notwithstanding). Issues are not a support forum; for that we have Discussions or the Matrix/Gitter channel.

**Comment by colepeters** [comment]
> Sorry if I sounded rude. Both neovim and nvim-treesitter are under heavy development and the API will be actively changed. If you experience any problems in the future after the update, you may be able to fix them by checking [this](https://github.com/neovim/neovim/issues/14090) issue.

Thanks, that's super helpful!

**Comment by colepeters** [comment]
> No, I will not reopen this, because this is not an issue in this project that needs to be fixed; it's an intentional breaking change (valid complaints about how this change could have been handled with more resources notwithstanding). Issues are not a support forum; for that we have Discussions or the Matrix/Gitter channel.

@clason I genuinely believed this was an issue with Treesitter, which is why I opened an issue rather than started a discussion. I was clearly incorrect and I realized and admitted that before your reply. Sometimes these breaking changes do not have obvious causes to end users who don't have as much information about interdependencies and other contexts as you do. Again, I recognize issues that end up being user errors can be time sucks for maintainers, and I appreciate your work, but this kind of hostility is really not warranted.
